static/test/javascripts/spec/common/discussion/recommendation.spec.js

```diff
@@ -1,17 +1,28 @@
 define([
     'common/utils/$',
-    'common/modules/discussion/upvote'
+    'helpers/injector'
 ],
 function (
     $,
-    upvote
+    Injector
 ) {
     describe('Recommendations of comments', function () {
         var discussionApi = {
             recommendComment: function () {}
-        };
+        },
+        injector = new Injector(), config, upvote;
+
+        beforeEach( function (done) {
+
+            injector.require(['common/modules/discussion/upvote', 'common/utils/config'], function() {
```

Comment:
I would have use dependency injection to pass `config` to the upvote module, which is a fancy way of saying

```js
function handle (target, container, user, discussionApi) {}
// becomes
function handle (target, container, user, discussionApi, config) {}
// or even
function handle (target, container, user, discussionApi, allowAnonymous) {}
// where
function allowAnonymous () { // defined in loader.js
   return config.switches.discussionAllowAnonymousRecommendsSwitch;
}
```
makes your tests straightforward. I've never liked `injector` but that's a personal thing, I don't understand how it works.